qutebrowser/browser/greasemonkey.py

```diff
@@ -356,40 +399,73 @@ def _add_script_with_requires(self, script, quiet=False):
         """
         # See if we are still waiting on any required scripts for this one
         for dl in self._in_progress_dls:
-            if dl.requested_url in script.requires:
+            if dl.requested_url in script.requires or \
+                    dl.requested_url in script.resources:
                 return False
 
         # Need to add the required scripts to the IIFE now
         for url in reversed(script.requires):
-            target_path = self._required_url_to_file_path(url)
+            target_path = self._required_url_to_file_path(
+                _RequirementType.requires, url)
             log.greasemonkey.debug(
                 "Adding required script for {} to IIFE: {}"
                 .format(script.name, url))
             with open(target_path, encoding='utf8') as f:
                 script.add_required_script(f.read())
 
+        for resource in script.resources:
+            target_path = self._required_url_to_file_path(
+                _RequirementType.resources, resource.url)
+            log.greasemonkey.debug(
+                "Adding required resource for {} to IIFE: {}"
+                .format(script.name, resource.url))
+            with open(target_path, encoding='utf8') as f:
+                script.add_resource(resource.name, f.read())
+
         self._add_script(script)
         if not quiet:
             self.scripts_reloaded.emit()
         return True
 
-    def _get_required_scripts(self, script, force=False):
-        required_dls = [(url, self._required_url_to_file_path(url))
-                        for url in script.requires]
-        if not force:
-            required_dls = [(url, path) for (url, path) in required_dls
-                            if not os.path.exists(path)]
+    def _get_requirements(self, script, force=False):
+        all_downloads = [
+            (url, self._required_url_to_file_path(
+                _RequirementType.requires, url)) for url in script.requires
+        ]+[
+            (resource.url, self._required_url_to_file_path(
+                _RequirementType.resources, resource.url))
+            for resource in script.resources]
+
+        required_dls = []  # type: typing.List[typing.Tuple[str, QUrl, str]]
+        for url, path in all_downloads:
+            url_ = QUrl(url)
+            file_exists = os.path.exists(path)
+
+            if url_.isRelative():
+                if not file_exists:
+                    message.error("Missing relative resource: {} for script {}"
```

Comment:
The user can copy the resource manually into the resources/ folder. Unless qutebrowser has a userscript manager it would not know where to download relative resource.